src/controllers/userProfileController.js

```diff
@@ -269,8 +269,9 @@ const userProfileController = function (UserProfile) {
         || req.body.requestor.requestorId === userid
       )
     );
+    const canUserToggleInvisibility = await hasPermission(req.body.requestor, 'toggleInvisibility');
+    if (!isRequestorAuthorized && !canUserToggleInvisibility) {
```

Comment:
The two conditions need not to be coupled. First check for request authorization, if not, stop progressing. Then check for invisibility toggle permission. The current code will fail to guard if users don't have authorization to make PUT requests but have permission to toggle invisibility. The new condition should follow the same logic like on line 280 with the 'addDeleteEditOwners' permission.